### PR TITLE
fix: add oneof fields to generated protoplus init

### PR DESCRIPTION
**Pull request by crwilcox**
Fixes: https://github.com/googleapis/gapic-generator-python/issues/484

### CODE REVIEW

**Comment by crwilcox** [comment]
It seems perhaps the data for oneofs may not be captured?

The template ought to be right provided oneof makes it to the field.

https://github.com/googleapis/gapic-generator-python/blob/920e419d4fb827f4cb154d0a38ffab83c19a0675/gapic/schema/api.py#L471

https://github.com/googleapis/gapic-generator-python/blob/920e419d4fb827f4cb154d0a38ffab83c19a0675/gapic/schema/api.py#L782

**Comment by BenRKarl** [comment]
@crwilcox is the issue here that oneof fields aren't present at all in your generated classes? If so I think we have the same problem for Ads, if a change is made here can we also change the ads templates? See [here](https://github.com/googleapis/gapic-generator-python/blob/master/gapic/ads-templates/%25namespace/%25name/%25version/%25sub/types/_message.py.j2#L46)

**Comment by crwilcox** [comment]
@BenRKarl For sure. Thanks for pointing that out. Still need to figure out piping the oneof data down to the field I think.

File: gapic/schema/api.py [hunk 1]
```diff
@@ -804,6 +844,7 @@ def _load_message(self,
                 address=address,
                 documentation=self.docs.get(path, self.EMPTY),
             ),
+            oneofs=oneofs,
```
Comment (crwilcox) [inline gapic/schema/api.py:847]: @software-dov  I didn't use this anywhere, but oneofs are part of a message so I thought having it here may serve useful

File: gapic/schema/api.py [hunk 2]
```diff
@@ -586,6 +615,10 @@ def _get_fields(self,
         # `_load_message` method.
         answer: Dict[str, wrappers.Field] = collections.OrderedDict()
         for field_pb, i in zip(field_pbs, range(0, sys.maxsize)):
+            oneof_name = None
+            if getattr(field_pb, 'oneof_index', -1) >= 0 and oneofs:
```
Comment (lukesneeringer) [inline gapic/schema/api.py:619]: Oh goodness, protobuf usability at its finest. 🤣

File: gapic/schema/api.py [hunk 3]
```diff
@@ -556,14 +556,43 @@ def _load_children(self,
...
+            path (Tuple[int]): The source location path thus far, as
+                understood by ``SourceCodeInfo.Location``.
+
+        Returns:
+            Mapping[str, ~.wrappers.Oneof]: A ordered mapping of
+                :class:`~.wrappers.Oneof` objects.
+        """
+        # Iterate over the oneofs and collect them into a dictionary.
+        answer: Dict[str, wrappers.Oneof] = collections.OrderedDict()
+        for oneof_pb, i in zip(oneof_pbs, range(0, sys.maxsize)):
```
Comment (software-dov) [inline gapic/schema/api.py:579]: Style nit: prefer
```python
answer = collections.OrderedDict(
    (oneof_pb.name, wrappers.Oneof(oneof_pb=oneof_pb))
    for i, oneof_pb in enumerate(oneof_pbs)
)
```

File: gapic/schema/api.py [hunk 4]
```diff
@@ -556,14 +556,43 @@ def _load_children(self,
...
+                oneof_pb=oneof_pb,
+            )
+
+        # Done; return the answer.
+        return answer
+
     def _get_fields(self,
                     field_pbs: Sequence[descriptor_pb2.FieldDescriptorProto],
                     address: metadata.Address, path: Tuple[int, ...],
+                    oneofs=None
```
Comment (software-dov) [inline gapic/schema/api.py:590]: Type annotations nit :P

File: gapic/schema/api.py (hunk 2 above)
Comment (software-dov) [reply gapic/schema/api.py:619]: 🤮

File: gapic/schema/api.py (hunk 1 above)
Comment (software-dov) [reply gapic/schema/api.py:847]: Can't hurt.

**Pull request merged**